### PR TITLE
[GEN]:  Introduce GEN dialect in Triton  - part 2

### PR DESCRIPTION
**Pull request by etiotto**
Create the dialect conversion pass to convert GEN operation into LLVM dialect operations. The following GEN operations are handled in this patch:
  - ThreadIdXOp, ThreadIdYOp, ThreadIdZOp
  - BlockIdXOp, BlockIdYOp, BlockIdZOp
  - BlockDimXOp, BlockDimYOp, BlockDimZOp
  - GridDimXOp, GridDimYOp, GridDimZOp
  - BarrierOp, SubGroupShuffleOp

### CODE REVIEW

File: third_party/intel/lib/GENToLLVM/GENToLLVMPass.cpp [hunk 1]
```diff
@@ -0,0 +1,422 @@
+//===- GENToLLVM.cpp - GEN to LLVM dialect conversion ---------------------===//
```
Comment (etiotto) [inline third_party/intel/lib/GENToLLVM/GENToLLVMPass.cpp:1]: GENToLLVM -> GENToLLVMPass

File: third_party/intel/include/GENToLLVM/GENToLLVMPass.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+class DialectRegistry;
+class LLVMTypeConverter;
+class RewritePatternSet;
+class Pass;
+
+#define GEN_PASS_DECL_CONVERTGENTOLLVM
+#include "intel/include/GENToLLVM/Passes.h.inc"
+
+namespace triton {
```
Comment (whitneywhtsang) [inline third_party/intel/include/GENToLLVM/GENToLLVMPass.h:24]:
```suggestion
namespace triton {

#define GEN_PASS_DECL
#include "intel/include/GENToLLVM/Passes.h.inc"
```

**Review by whitneywhtsang — COMMENTED** [review]
Please add test cases.

File: third_party/intel/include/GENToLLVM/GENToLLVMPass.h (hunk 2 above)
Comment (etiotto) [reply third_party/intel/include/GENToLLVM/GENToLLVMPass.h:24]: I prefer to follow the style of dialects in tree (e.g. arith dialect).

Comment (whitneywhtsang) [reply third_party/intel/include/GENToLLVM/GENToLLVMPass.h:24]: ok, I was comparing to other conversion passes in Triton.

**Comment by etiotto** [comment]
Test case added

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**